api/src/main/java/com/codeforcommunity/api/IReservationProcessor.java

```diff
@@ -0,0 +1,37 @@
+package com.codeforcommunity.api;
+
+import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.reservation.*;
+
+public interface IReservationProcessor {
+
+  /**
+   * Marks the given block as reserved for the given user/team by creating a new entry in the
+   * reservation table
+   */
+  void makeReservation(JWTData userData, MakeReservationRequest makeReservationRequest);
+
+  /**
+   * Marks the given block as complete for the given user/team by creating a new entry in the
+   * reservation table
+   */
+  void completeReservation(JWTData userData, CompleteReservationRequest completeReservationRequest);
```

Comment:
We should probably combine CompleteResrvationRequest and MakeReservationRequest as well... Something like BlockIDOptionalTeamIdRequest???

Reply:
Feel like this is clearer for now and might receive changes when teams is implemented, don't want to have to undo it.